### PR TITLE
Add pagination speakers module

### PR DESCRIPTION
**Pull request by auliafaizahr**
**Title: Add Pagination to Speaker List and Adjust Layout**

**Description**
Feature Request: As an admin user, I want to manage Speakers for our Events.

**Related Problem:**
The list of speakers in speaker module is expected to grow significantly, making it difficult to maintain and navigate.

**Solution:**
This pull request adds pagination to the speaker list to improve usability. The solution utilizes the Pagy gem for pagination and Bootstrap for styling.

**Changes:**

image

- [x] Added pagination using pagy gem to the speaker list with 15 elements per page.
- [x] Moved the "Create Speaker" button to share the same row with the title.
- [x] Adjusted the layout to position the pagination controls on top of the view.

**Related Issues:** https://github.com/wnbrb/wnb-rb-site/issues/184

Waiting for review the changes and provide feedback from reviewer, thankyou.

### CODE REVIEW

**Review by luciagirasoles — CHANGES_REQUESTED** [review]
Congratulations for the work done here 🎉! It looks so good! we still need to improve in this PR is how we display the composition of this component in this page. Let's try to keep the same structure as it is in the Event List for Desktop and Mobile screens. (Images sent on Slack)

File: Gemfile.lock [hunk 1]
```diff
@@ -377,31 +393,24 @@ GEM
...
+    zeitwerk (2.6.17)
 
 PLATFORMS
-  arm64-darwin-21
-  arm64-darwin-22
-  arm64-darwin-23
-  x86_64-darwin-19
-  x86_64-darwin-20
-  x86_64-darwin-21
-  x86_64-darwin-22
```
[the remark this answers is not in the document]
Comment (auliafaizahr) [reply Gemfile.lock]: Restored, pls kindly check.

**Review by luciagirasoles — APPROVED** [review]
Thank you for the upgrades and for supporting the community.

**Comment by luciagirasoles** [comment]
closes #238

**Pull request merged**